decision/engine.go

```diff
@@ -524,9 +766,23 @@ func (e *Engine) MessageSent(p peer.ID, m bsmsg.BitSwapMessage) {
 	l.lk.Lock()
 	defer l.lk.Unlock()
 
+	// Remove sent blocks from the want list for the peer
 	for _, block := range m.Blocks() {
 		l.SentBytes(len(block.RawData()))
-		l.wantList.Remove(block.Cid())
+		l.wantList.RemoveType(block.Cid(), pb.Message_Wantlist_Block)
+	}
+
+	// Remove sent block presences from the want list for the peer
+	for _, bp := range m.BlockPresences() {
```

Comment:
What if I send the actual block because it's small enough?

Reply:
I think it will actually be ok:
- The engine receives want-have for a block that is small enough that we just respond with the block itself (instead of a HAVE message)
  - The engine adds want-have to the wantlist
  - The engine adds want-block to the request queue
- When the request is popped off the request queue
  - The engine sends the block
  - The engine calls
    `l.wantList.RemoveType(block.Cid(), pb.Message_Wantlist_Block)`
    This will remove from the wantlist any want-have or want-block with that cid